src/main/src/main/java/org/geoserver/catalog/CoverageReaderInputObjectConverter.java

```diff
@@ -34,4 +33,22 @@
      * @return an {@link Optional} containing the converted value.
      */
     Optional<T> convert(Object input, @Nullable CoverageInfo coverageInfo, @Nullable Hints hints);
+
+    /**
+     * This method inspects the provided input object in an attempt to convert it to a custom class.
+     * Any of the accompanying method parameters may optionally be used to better inform the
+     * decision making logic. If an implementation does not support conversion for the given input
+     * object, the method should return an empty {@link Optional}.
+     *
+     * @param input The input object.
+     * @param coverageStoreInfo The coverage store metadata, may be <code>null</code>.
+     * @param coverageInfo The grid coverage metadata, may be <code>null</code>.
+     * @param hints Hints to use when loading the coverage, may be <code>null</code>.
+     * @return an {@link Optional} containing the converted value.
+     */
+    Optional<T> convert(
+            Object input,
+            @Nullable CoverageStoreInfo coverageStoreInfo,
```

Comment:
Looking at the ImageMosaic CogConfiguration, which hosts basically a CogConfig + username/password, wondering if the username/pwd could also be shared in CogConfig... could one have default credentials stored just once? I believe the AWS client can also read env variables, in other words, the idea that one uses fixed credentials for the entire app is not so strange in the cloud. Mind, just thinking out loud... and going there would require to have the possibility to override.

Reply:
Right, for some reason I was getting a null storeInfo at the early stage of development... Updated.
About CogSettings, it may end up having a couple of more parameters in the future.

The S3 RangeReader implementation actually can read credentials from those IIO_AWS_  System properties too.
I'm going to update that section of the doc.

Reply:
With interactive testing, found out that the store is necessary, because the coveage info can be missing, while creating the coveage reader for the first one, while configuring the store from the UI (e.g, a time when the eventual CoverageInfo has not yet been created).

Reply:
Restored.